### PR TITLE
meta: separate reader and mutator

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #54436

Problem Summary:

### What changed and how does it work?
meta.NewMeta is meaningless, we abstract a meta Reader and separate it out of Mutator
### Check List

Tests 

- [x] Unit test. existing ut should cover it, we mostly only rename structs
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/backup/client.go [hunk 1]
```diff
@@ -936,37 +936,6 @@ func BuildBackupSchemas(
 	return nil
 }
 
-// BuildFullSchema builds a full backup schemas for databases and tables.
-func BuildFullSchema(storage kv.Storage, backupTS uint64, fn func(dbInfo *model.DBInfo, tableInfo *model.TableInfo)) error {
```
Comment (D3Hunter) [inline br/pkg/backup/client.go:940]: useless now

**Review by lcwangchao — APPROVED** [review]

**Review by tiancaiamao — APPROVED** [review]

**Review by 3pointer — APPROVED** [review]

**Review by lance6716 — APPROVED** [review]

**Review by yudongusa — APPROVED** [review]
LGTM

**Pull request merged**